drivers/hmis/app/graphql/types/hmis_schema/referral_posting.rb

```diff
@@ -41,12 +41,25 @@ class HmisSchema::ReferralPosting < Types::BaseObject
     field :referred_from, String, null: false
     field :unit_type, HmisSchema::UnitTypeObject, null: false
 
+    # If this posting has been accepted, this is the enrollment for the HoH at the enrolled household.
+    # This enrollment is NOT necessarily the same as the `hoh_name`, because the HoH may have changed after
+    # posting was accepted.
+    field :hoh_enrollment, HmisSchema::Enrollment, null: true
+
     def hoh_name
       object.referral.household_members
         .detect(&:self_head_of_household?)
         &.client&.brief_name
     end
 
+    def hoh_enrollment
+      return unless object.household_id
+
+      Hmis::Hud::Enrollment.viewable_by(current_user).
+        where(household_id: object.household_id).
+        heads_of_households.first
```

Comment:
As a hacky way to reduce n+1 queries, I used a bunch of preload statements here [drivers/hmis/app/graphql/types/hmis_schema/has_referral_postings.rb](https://github.com/greenriver/hmis-warehouse/blob/gig/hoh-referral-enrollment/drivers/hmis/app/graphql/types/hmis_schema/has_referral_postings.rb#L30). 

I guess that doesn't work here though since we can't do a visibility check without the query. We should probably circle back replace all that with data loaders when we figure the pagination issue

Reply:
Interesting, yeah. Tracking that [here](https://www.pivotaltracker.com/n/projects/2591838/stories/185336364).

In this case, maybe we should define a `has_many :enrollments` or `has_one :hoh_enrollment` on the ReferralPosting, so it can be preloaded. The relationship is not _just_ based on household ID, though, it needs to be by data_source_id as well. (Household ID is only unique along with data source). We'll need to store data source id in the referral posting table in order to make that work, though. I think that makes sense to do.

Reply:
That sounds reasonable. Using a composite key  of data source and enrollment_id is also something we could encapsulate in a data loader if we are moving there in the future.

Just for my own understanding, do we need to use household_id? Is that too reference other records beyond enrollment? My very naive idea would be to replace the posting.household_id with something like this:
```
add_reference :hmis_external_referral_postings, :hoh_enrollment, foreign_key: { to_table: 'Enrollment'}
```

Reply:
The reason to use Household ID (rather than Enrollment) is that the household composition might change over time, and the original hoh's enrollment could get removed from the household. Unlikely, but possible. For example:

1. Referral is accepted into program, member 1,2,3 are Enrolled in the household, with 1 as the HoH
2. User changes member 2 to be the HoH
3. User removes member 1 from household (which deletes their Enrollment, since it's still in "incomplete" state)
4. User "finalizes" enrollments by submitting member 2 and 3's intake assessments
5. We still want to be able to link this household (consisting of members 2 and 3) back to the referral that it originated from.

Technically it could be even more extreme - ALL the original referred members were removed, and new ones added. The only thing linking that household together is the Household ID. 


We already do use a composite key with data source in a _lot_ of places, that's what the `hmis_relation` helper handles. I don't really follow how that interacts with data loading, though.

Reply:
https://github.com/greenriver/hmis-warehouse/pull/3131/commits/e9db32ad5bc915a15ee33cf787c31ab5168dfeea

Reply:
Got it, that makes sense, thanks for the explanation!